variable/include/scipp/variable/bucket_model.h

```diff
@@ -77,6 +77,10 @@ template <class T> class DataModel<bucket<T>> : public VariableConcept {
     return {index_values(base), m_dim, m_buffer};
   }
 
+  scipp::index dtype_size() const {
+    return sizeof(T);
```

Comment:
So `T` may be `Variable`, `DataArray`, or `Dataset`. Is returning `sizeof(T)` useful?

Reply:
No this is not a useful function for bucketed data. It has to implement something as the function is in the interface though. Would just returning 1 or 0 so it's clear this is not useful info be better?

Reply:
Maybe it should just return the size of the index/range dtype?